### PR TITLE
[KernelGen] Fix --ref cpu cross-device tensor comparison

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
## Summary
- Fix `_maybe_move_to_cpu()` in `src/flag_gems/testing/__init__.py` to handle cross-device tensor comparison by moving both tensors to CPU when they are on different devices
- Add root-level `conftest.py` to patch `gems_assert_close` / `gems_assert_equal` for `--ref cpu` compatibility when tests compute reference on GPU

Fixes internal issues #411, #412, #413 (`act_quant_triton`, `addmm_dtype`, `addmm_dtype_out` accuracy test failures under `--ref cpu` mode).

## Test plan
- [x] `pytest -m "act_quant_triton" tests/ --ref cpu -vs` — 112 passed
- [x] `pytest -m "addmm_dtype" tests/ --ref cpu -vs` — 3 passed
- [x] `pytest -m "addmm_dtype_out" tests/ --ref cpu -vs` — 3 passed
- [x] All 114 addmm-related tests pass with and without `--ref cpu`
- [x] Benchmarks pass

## Issue
- WEEKTEST-411
- WEEKTEST-412
- WEEKTEST-413

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+tests/accuracy_utils.to_cpu fails for operators whose test manually
+computes the reference on CPU and then moves it to the accelerator
+device.  We relax the assertion to instead move *ref* to CPU, which
+is the safe and correct thing to do for comparison purposes.
+"""
+
+import torch
+
+
+def pytest_configure(config):
```
Comment (tengqm) [inline conftest.py]: don't do this.

If you are confident that this is the right approach for accuracy testing, update the tests/conftest.py directly. We don't want any .py files in the root directory.

**Review by tengqm — APPROVED** [review]
